DataFormats/FWLite/src/DataGetterHelper.cc

```diff
@@ -131,20 +133,22 @@ namespace fwlite {
         //obj.destruct();
         ////END OF WORK AROUND
 
-        TTreeCache* tcache = dynamic_cast<TTreeCache*> (branchMap_->getFile()->GetCacheRead());
+        if (tcUse_) {
+            TTreeCache* tcache = dynamic_cast<TTreeCache*> (branchMap_->getFile()->GetCacheRead());
 
-        if (nullptr == tcache) {
-            iData.branch_->GetEntry(eventEntry);
-        } else {
-            if (!tcTrained_) {
-                tcache->SetLearnEntries(100);
-                tcache->SetEntryRange(0, tree_->GetEntries());
-                tcTrained_ = true;
+            if (nullptr != tcache) {
+                if (!tcTrained_) {
+                    tcache->SetLearnEntries(100);
+                    tcache->SetEntryRange(0, tree_->GetEntries());
+                    tcTrained_ = true;
+                }
+                tree_->LoadTree(eventEntry);
             }
-            tree_->LoadTree(eventEntry);
-            iData.branch_->GetEntry(eventEntry);
-       }
-       iData.lastProduct_=eventEntry;
+        }
+        branchAccFoo_(iData.branch_);
```

Comment:
In fworks we do need to add / remove branches to the tree cache as collections get added and removed into the list. Additionally, when detailed-view is opened, other collections can be added ... that's why I need callback from event/data helper.

I don't know who else could use this. But in any case you could reduce learn entries and probably do learn-prefill of all branches.